### DISCUSSION TITLE
请求提供更多鼠标移动方式或者库

### DISCUSSION THREAD

**Discussion by Passer1072** [Passer1072/RookieAI_yolov8#25] [Q&A]
如果你有更好的鼠标移动方式或者库，欢迎在这里讨论！

**Comment by whilemoyan** [comment]
![image](https://github.com/Passer1072/RookieAI_yolov8/assets/137123467/bf772840-5e99-4c7c-943b-1a2338f35298)
 调用按键魔盒移动

**Reply by Passer1072** [reply]
> ![image](https://private-user-images.githubusercontent.com/137123467/345981210-bf772840-5e99-4c7c-943b-1a2338f35298.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNTE0MTQsIm5iZiI6MTcyMDE1MTExNCwicGF0aCI6Ii8xMzcxMjM0NjcvMzQ1OTgxMjEwLWJmNzcyODQwLTVlOTktNGM3Yy05NDNiLTFhMjMzOGYzNTI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQwMzQ1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NzAzMGRkODg0MzY3ZThlZmFlMjMzZDY3MGYyOThiNzFhNmU4YWNlNWVlNDljMjgwNmVhYjIyZmY5ODcwODU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qcUcA0o9kj2_0L_Ngtx0k7whDOiNAcIu-fEj8rYwhyw) 调用按键魔盒移动

好的，稍后会去采购一个KMBOX B来开发

**Reply by kujoykarendesu** [reply]
> > ![image](https://private-user-images.githubusercontent.com/137123467/345981210-bf772840-5e99-4c7c-943b-1a2338f35298.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNTE0MTQsIm5iZiI6MTcyMDE1MTExNCwicGF0aCI6Ii8xMzcxMjM0NjcvMzQ1OTgxMjEwLWJmNzcyODQwLTVlOTktNGM3Yy05NDNiLTFhMjMzOGYzNTI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQwMzQ1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NzAzMGRkODg0MzY3ZThlZmFlMjMzZDY3MGYyOThiNzFhNmU4YWNlNWVlNDljMjgwNmVhYjIyZmY5ODcwODU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qcUcA0o9kj2_0L_Ngtx0k7whDOiNAcIu-fEj8rYwhyw) 调用按键魔盒移动
> 
> 好的，稍后会去采购一个KMBOX B来开发

大佬，我现在也想学学这一类的正好想来问问有没有调用kmbox的 不过好像现在主流的好像是kmboxnet？ 我见说是kmbox b那种是波特率传输好像会检测？ 我不太清楚具体的 大佬有什么交流群吗

**Reply by Passer1072** [reply]
> > > ![image](https://private-user-images.githubusercontent.com/137123467/345981210-bf772840-5e99-4c7c-943b-1a2338f35298.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNTE0MTQsIm5iZiI6MTcyMDE1MTExNCwicGF0aCI6Ii8xMzcxMjM0NjcvMzQ1OTgxMjEwLWJmNzcyODQwLTVlOTktNGM3Yy05NDNiLTFhMjMzOGYzNTI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQwMzQ1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NzAzMGRkODg0MzY3ZThlZmFlMjMzZDY3MGYyOThiNzFhNmU4YWNlNWVlNDljMjgwNmVhYjIyZmY5ODcwODU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qcUcA0o9kj2_0L_Ngtx0k7whDOiNAcIu-fEj8rYwhyw) 调用按键魔盒移动
> > 
> > 
> > 好的，稍后会去采购一个KMBOX B来开发
> 
> 大佬，我现在也想学学这一类的正好想来问问有没有调用kmbox的 不过好像现在主流的好像是kmboxnet？ 我见说是kmbox b那种是波特率传输好像会检测？ 我不太清楚具体的 大佬有什么交流群吗

没有群也暂时没有建群的打算，关于kmbox因为我也没有用过kmbox的任何产品，得到时候买个kmbox net来看看。

**Reply by kujoykarendesu** [reply]
> > > > ![image](https://private-user-images.githubusercontent.com/137123467/345981210-bf772840-5e99-4c7c-943b-1a2338f35298.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNTE0MTQsIm5iZiI6MTcyMDE1MTExNCwicGF0aCI6Ii8xMzcxMjM0NjcvMzQ1OTgxMjEwLWJmNzcyODQwLTVlOTktNGM3Yy05NDNiLTFhMjMzOGYzNTI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQwMzQ1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NzAzMGRkODg0MzY3ZThlZmFlMjMzZDY3MGYyOThiNzFhNmU4YWNlNWVlNDljMjgwNmVhYjIyZmY5ODcwODU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qcUcA0o9kj2_0L_Ngtx0k7whDOiNAcIu-fEj8rYwhyw) 调用按键魔盒移动
> > > 
> > > 
> > > 好的，稍后会去采购一个KMBOX B来开发
> > 
> > 大佬，我现在也想学学这一类的正好想来问问有没有调用kmbox的 不过好像现在主流的好像是kmboxnet？ 我见说是kmbox b那种是波特率传输好像会检测？ 我不太清楚具体的 大佬有什么交流群吗
> 
> 没有群也暂时没有建群的打算，关于kmbox因为我也没有用过kmbox的任何产品，得到时候买个kmbox net来看看。

好的，大佬这类学习视频去哪能找到呢 我现在也想学习学习这些。不过好像没有教学获取途径

**Reply by Passer1072** [reply]
> 
> 好的，大佬这类学习视频去哪能找到呢 我现在也想学习学习这些。不过好像没有教学获取途径

呃这种一般没有完整系统性的学习资料更多是参考学习别人的源码，理解了原理后根据自己想法来写，期间任何不懂的都可以问Ai。先写一个能跑的程序UI都不需要的后面再一点点添加东西。
这是我最早参考的资料：[How I made an AI to aim for me!](https://youtu.be/lzJtEYA71jw?si=dS4K17fS6hPU6Icq)，和它的[源码](https://github.com/OfficialProtonDev/ML-Aimbot-Halo-Infinite)。
读懂源码后我进行了一些重构就有了早期的YOLOv5的版本：[RookieAI](https://github.com/Passer1072/RookieAI)
后来由于不满足它源码的拓展性加上它本身写的很乱且不支持YOLOv8就新开了现在这个项目，根据之前从YOLOv5版本积累的经验加上自己的想法和AI的帮助写出了v8最早的控制台版本没有GUI，再后来稍微完善细节体验后为了方便调参加上了tkinterGUI就有了v8最早带GUI用着还行的版本[RookieAI_YOLOv8_v1.0_oldGUI](https://github.com/Passer1072/RookieAI_yolov8/tree/main/historic_version/RookieAI_YOLOv8_v1.0_oldGUI)
再后来换为美化版的tkinterGUI库[CustomTkinter](https://customtkinter.tomschimansky.com/)就有了现在还看得过去的UI样子
之后就根据日常使用体验和需求不断小修小补加新功能。
这就是我开发这个项目的大致经历了，历史版本都在[历史记录](https://github.com/Passer1072/RookieAI_yolov8/tree/main/historic_version)中，代码的注释也写的挺多的还是比较容易看懂，希望这些对你有所帮助！哪里有问题可以一起交流分享！

**Reply by kujoykarendesu** [reply]
> > 好的，大佬这类学习视频去哪能找到呢 我现在也想学习学习这些。不过好像没有教学获取途径
> 
> 呃这种一般没有完整系统性的学习资料更多是参考学习别人的源码，理解了原理后根据自己想法来写，期间任何不懂的都可以问Ai。先写一个能跑的程序UI都不需要的后面再一点点添加东西。 这是我最早参考的资料：[How I made an AI to aim for me!](https://youtu.be/lzJtEYA71jw?si=dS4K17fS6hPU6Icq)，和它的[源码](https://github.com/OfficialProtonDev/ML-Aimbot-Halo-Infinite)。 读懂源码后我进行了一些重构就有了早期的YOLOv5的版本：[RookieAI](https://github.com/Passer1072/RookieAI) 后来由于不满足它源码的拓展性加上它本身写的很乱且不支持YOLOv8就新开了现在这个项目，根据之前从YOLOv5版本积累的经验加上自己的想法和AI的帮助写出了v8最早的控制台版本没有GUI，再后来稍微完善细节体验后为了方便调参加上了tkinterGUI就有了v8最早带GUI用着还行的版本[RookieAI_YOLOv8_v1.0_oldGUI](https://github.com/Passer1072/RookieAI_yolov8/tree/main/historic_version/RookieAI_YOLOv8_v1.0_oldGUI) 再后来换为美化版的tkinterGUI库[CustomTkinter](https://customtkinter.tomschimansky.com/)就有了现在还看得过去的UI样子 之后就根据日常使用体验和需求不断小修小补加新功能。 这就是我开发这个项目的大致经历了，历史版本都在[历史记录](https://github.com/Passer1072/RookieAI_yolov8/tree/main/historic_version)中，代码的注释也写的挺多的还是比较容易看懂，希望这些对你有所帮助！哪里有问题可以一起交流分享！

十分感谢！！！！ 我也去按照这个思路走走

**Comment by Borei99** [comment]
叛逆魔盒，幽灵键鼠，无涯键鼠盒子都还可以

**Reply by Passer1072** [reply]
> 叛逆魔盒，幽灵键鼠，无涯键鼠盒子都还可以

这三个包括易键鼠，飞易来文盒其实都是同样的东西没有什么特别功能，这一类我目前根据飞易来文盒开发，应该是兼容其他几个的或者小改下什么肯定兼容的，故不再考虑这几个。
